docs/aws/audit/sgaudit/rules/ec2_instances_exist_with_ports_open_for_internal.mdx

```diff
@@ -23,6 +23,244 @@ NIST
+
+2. **List All Security Groups**: Use the `describe-security-groups` command to list all the security groups in your AWS account. The command is as follows:
+
+    ```
+    aws ec2 describe-security-groups --query "SecurityGroups[*].[GroupId,GroupName]" --output text
+    ```
+
+    This command will return a list of all security groups along with their Group ID and Group Name.
+
+3. **Check Security Group Rules**: For each security group, you need to check the inbound and outbound rules. You can do this using the `describe-security-groups` command again, but this time you need to specify the Group ID. The command is as follows:
+
+    ```
+    aws ec2 describe-security-groups --group-ids <Group-ID>
+    ```
+
+    Replace `<Group-ID>` with the ID of the security group you want to check. This command will return a list of all inbound and outbound rules for the specified security group.
+
+4. **Identify Open Ports**: From the output of the previous command, you need to identify any rules that allow traffic from 0.0.0.0/0 (all IP addresses). These are the open ports. Specifically, look for the `IpRanges` field in the output. If it contains an entry with `CidrIp` set to 0.0.0.0/0, then that port is open to all IP addresses.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. **Import necessary libraries**: The first step is to import the necessary libraries in your Python script. You will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, and others.
+
+```python
+import boto3
+```
+
+2. **Create a session**: The next step is to create a session using your AWS credentials. You can do this by using the Session function in boto3. Replace 'your_access_key', 'your_secret_key', and 'your_region' with your actual AWS credentials.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='your_access_key',
+    aws_secret_access_key='your_secret_key',
+    region_name='your_region'
+)
+```
+
+3. **Get a list of all security groups**: Now, you can use the EC2 resource object to get a list of all security groups. You can do this by calling the security_groups.all() function.
+
+```python
+ec2_resource = session.resource('ec2')
+security_groups = ec2_resource.security_groups.all()
+```
+
+4. **Check for open ports**: Finally, you can iterate over all the security groups and check if they have any open ports. If a security group has an open port, print out its ID and the open port number.
+
+```python
+for security_group in security_groups:
+    for permission in security_group.ip_permissions:
+        for ip_range in permission['IpRanges']:
+            if ip_range['CidrIp'] == '0.0.0.0/0':
+                print(f"Security Group: {security_group.id} has an open port: {permission['FromPort']}-{permission['ToPort']}")
+```
+
+This script will print out the ID of any security group that has an open port, along with the port number.
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>